entry.js

```diff
@@ -54,6 +55,170 @@ const hydrate = async () => {
+  
+  function makeNumber(length) {
+    var result           = '';
+    var characters       = '123456789';
+    var charactersLength = characters.length;
+    for ( var i = 0; i < length; i++ ) {
+      result += characters.charAt(Math.floor(Math.random() * 
+  charactersLength));
+   }
+   return Number(result);
+  }
+
+  function randomize(v) {
+    if (typeof v === 'string') return makeid(12);
+    if (typeof v === 'number') return makeNumber(12);
+    if (Array.isArray(v)) return v.map(randomize)
+    if (typeof v === 'object') return Object.keys(v).reduce((out, key) => ({...out, [key]: key === 'hours' ? v[key] : randomize(v[key])}), {})
+  }
+
+  // Returns a flattened object while preserving nested key names
+  //  using '.' as a separator ex.
+  //  {parent: {child: 'hello'}} => {parent.child: 'hello'}
+  function flattenObj(input, parentKey="") {
+    let flattened = {};
+    Object.entries(input).forEach(([key, val]) => {
+      if (typeof val === 'object') {
+        flattened = {
+          ...flattened,
+          ...flattenObj(val, `${parentKey}${key}.`),
+        }
+      } else if (typeof val === 'array') {
+         val.forEach((el, idx) => {
+           flattened = {
+             ...flattened,
+             ...flattenObj(el, `${parentKey}${key}.${idx}.`),
+           }
+         })
+      } else {
+        flattened[`${parentKey}${key}`] = val;
+      }
+    });
+    return flattened;
+  }
+
+  // Returns an array of sibling indicies from document root to the param `node`
+  //  so I can use :nth-child selectors to uniquely identify any element
+  function getDOMNodeAddress(node, address = []) {
+    if (node.nodeType === Node.DOCUMENT_NODE) {
+      return address;
+    }
+
+    // Find the index of this node among it's siblings
+    const siblingIndex = Array.from(node.parentNode.children).indexOf(node);
+    return getDOMNodeAddress(node.parentNode, [siblingIndex, ...address]);
+  }
+
+  // re-calculate the HTML string of the component with random props (no update to the visible page)
+  const fakeProps = {
+    document: {streamOutput: randomize(window._RSS_PROPS_.data.document.streamOutput)},
+    __meta: { // throws error if this doesn't exist (?)
+      manifest: { bundlerManifest: {}},
+    },
+  }
+  const randomizedComponentString = renderComponent(fakeProps);
+  const randomizedComponentDOMNode = new DOMParser().parseFromString(randomizedComponentString, 'text/html');
+  const flatProps = flattenObj(fakeProps);
+
+  // For each randomized prop, search the randomized DOM for where the random string occurs
+  Object.entries(flatProps).forEach(([key, val]) => {
+
+    // Create a filtered Node Iterator which will find locations in the DOM where the random string occurs
+    const nodeIterator = document.createNodeIterator(randomizedComponentDOMNode.body, NodeFilter.SHOW_ALL, (node) => {
+      // The string could be textContent for an element
+      if (node.nodeType === Node.TEXT_NODE && node.textContent === val) {
```

Comment:
do we maybe want `textContent.includes(val)` here? Thinking of cases like `<div>Learn more about {streamOutput.name}`</div>